src/io.rs

```diff
@@ -65,10 +63,12 @@ fn process_text<W: Write>(text: &str, mut out: W) -> Result<()> {
         )?;
     }
     let parser = Parser::new(tokens);
-    for err in parser.ast().errors() {
+    let ast = parser.ast();
+    for err in ast.errors() {
```

Comment:
Do we want to quit if there are AST errors? Evaluating a broken AST seems wrong.

Reply:
yes, coming in a future PR